File: Chapter04/4.2 Downward Dependency/BadWay/SomeRepository.cs
namespace BadWay
{
    // DataAccessLayer.cs 
    public class SomeRepository 
    {
        // This is the violation! A lower layer should not
        // know about an upper layer.
        private PresentationLayer _uiLayer = PresentationLayer.Instance;

        public void UpdateData(int id, string newData)
        {
            Console.WriteLine("(BAD) Saving data to database...");
            
            // VIOLATION! Calling upwards to the UI Layer
            _uiLayer.UpdateStatusLabel($"(BAD) Data {id} Saved!");
        }
    }
}

File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_2_downward_dependency/before/SomeRepository.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_2_downward_dependency.before;

import com.grokkingsoftwarearchitecture.chapter04.shared.LogManager;

/**
 * ARCHITECTURE WARNING: Upward Dependency Violation.
 * DataAccessLayer.java 
 */
public class SomeRepository {

    public void updateData(int id, String newData) {
        LogManager.info(SomeRepository.class, "(Before) Saving data to database...data: " + newData);
        
        // VIOLATION: Calling upwards to the UI Layer
        PresentationLayer.updateStatusLabel("(Before) Data " + id + " Saved!");
    }
}